js/main.js

```diff
@@ -41,3 +50,48 @@ function handleSearch(entry){
         entry.classList.add('hidden')
       }
     }
+
+const filterIcon=document.getElementById('filter-icon')
+const typeFilterContainer=document.getElementById('type-filter-container')
+filterIcon.addEventListener('click',()=>{
+  if(typeFilterContainer.style.display==='flex'){
+    typeFilterContainer.style.display='none'
+  }
+  else{
+    typeFilterContainer.style.display='flex'
+  }
+
+})
+
+const typeCheckboxes=document.querySelectorAll('.checkbox')
+function handleTypeFilter(event){
+  if(event.target.className==='checkbox'){
+    let counter=0
+    typeCheckboxes.forEach((value)=>{
+      if(value.checked===true){
```

Comment:
fyi: this is redundant. assuming `value.checked` returns a boolean, you can simplify to `if(value.checked)`